app/components/snapshot-timeline/component.js

```diff
@@ -0,0 +1,80 @@
+import Ember from 'ember';
+import ThrottledResize from 'ui/mixins/throttled-resize';
+
+const { get, set, computed } = Ember;
+
+export default Ember.Component.extend(ThrottledResize, {
+  tagName: 'div',
+  classNames: ['timeline-container'],
+  startDate: null,
+  endDate: null,
+  range: null,
+
+  init: function() {
+    this._super(...arguments);
+
+    if (!get(this, 'model')) {
+      set(this, 'classNames', []);
+    }
+
+  },
+
+  onResize: function() {
+    if (get(this, 'model')) {
+      this.plotSnapshots(this.get('startDate'), this.get('endDate'), this.get('range'), this.get('model'));
+    }
+  },
+
+  snapshots: computed('model', function() {
+    return this.plotSnapshots();
+  }),
+
+  plotSnapshots() {
+    let prevPlot = null;
+    let snapshots = get(this, 'model').sortBy('created');
+    let model = get(this, 'model');
+    let start = model[0].created;
+    let end   = model[model.length - 1].created;
+    let range = moment(end).diff(start, 'seconds');
+
+    snapshots.forEach((snapshot) => {
+      let myDate    = snapshot.created;
+      let fromStart = moment(myDate).diff(start, 'seconds');
+      let shift     = (fromStart/range) * 100;
+
+      if (prevPlot !== null && prevPlot >= 0 && shift !== 100) {
+        if ((shift - prevPlot) <= 3) {
```

Comment:
Make the `3` and `15` consts at the top of the file.  Does `-15px/2` get rounded the same way in all browsers?  Maybe it should just be explicitly 7.